pkg/api/validation/validation.go

```diff
@@ -2865,6 +2865,22 @@ func ValidateService(service *api.Service) field.ErrorList {
 		nodePorts[key] = true
 	}
 
+	// Check for duplicate Ports, considering (protocol,port) pairs
+	portsPath = specPath.Child("ports")
+	ports := make(map[api.ServicePort]bool)
+	for i, port := range service.Spec.Ports {
+		if port.Port == 0 {
```

Comment:
I thought port can be unspecified. That's incorrect. Thanks for pointing this out!